src/Emogrifier/Utilities/CssConcatenator.php

```diff
@@ -74,7 +74,7 @@ public function append(array $selectors, $declarationsBlock, $media = '')
             $lastRuleBlock->selectorsAsKeys += $selectorsAsKeys;
         } else {
             $hasSameSelectorsAsLastRule = $lastRuleBlock !== false
-                && static::hasEquivalentSelectors($selectorsAsKeys, $lastRuleBlock->selectorsAsKeys);
+                && self::hasEquivalentSelectors($selectorsAsKeys, $lastRuleBlock->selectorsAsKeys);
```

Comment:
My first thought was that I don't think any of these in `CssConcatenator` should be changed.  The proposal in #668 is to allow this class to be extended - there is even a visibility change of a property from `private` to `protected` proposed.  We should allow the methods to be overridden via late static binding.

However, on closer examination, I see that the particular methods are all `private`, so they can't be overridden (unless I'm missing something that wouldn't be the case if they were non-static methods), and there is currently no proposal in #668 to change their visibility.  (That may change, but we can revisit if it does.)

So in conclusion, I think this change is correct for consistency with the (current) methods' visibilities.